### ISSUE TITLE
Recipe Shortcode: Allow filtering of output_time

### ISSUE THREAD

**Issue by R33D3M33R** [Automattic/jetpack#21708]
### Impacted plugin

Jetpack

### What

Please allow filtering of output_time in Recipe Shortcode

### How

Dear developers,

the recipe shortcode output_time currently adds proper metadata to strings that strtotime recognizes. For localized strings, the metadata will thus be invalid. Filtering the return value would allow to replace the invalid $time_str with proper one.

Example
`return apply_filters('jetpack_recipes_output_time', $formatted_time, $time_str, $time_type)`

Best regards,
Andrej

**Comment by jeherve** [comment]
While a filter may help, I think it may be more beneficial for everyone if we were to update the way the output is built, and transform the localized time into one that's going to be able to be read by `strtotime`.

I believe we should be able to do this with some of the date / time tools made available in WP Core (see `WP_Locale` and related functions).

**Comment by jeherve** [comment]
Do you have a specific idea of how you would dynamically replace the strings? Would that be based on a hardcoded list of terms (localized terms for hours, minutes, ...)?

**Comment by R33D3M33R** [comment]
Yes, I would use static replacement arrays (hours, minutes). After replacement request review from Google Webmasters and fix any edge cases.
